nx/source/services/spl.c

```diff
@@ -94,6 +94,11 @@ void splExit(void) {
     return _splSrvExit(&g_splSrv, &g_splRefCnt);
 }
 
+Service* splGetServiceSession(void)
```

Comment:
spl should not return service sessions this way. It will break the muxing we do for < 4.x deprecated service vs >= 4.x modern services.

Use the _splGet*Srv accessors.

Reply:
Should those functions be put in the header and renamed to the `GetServiceSession` format?

Reply:
That's kind of awkward due to RsaService, which doesn't have any real service equivalent.

Reply:
So what do you suggest? Since I think it'd be awkward having a bunch of functions where all they do is just return what another function returns, and there doesn't seem to be `_splGetSslSrv` or `_splGetManuSrv` functions, which would be awkward for those respective services

Reply:
"where all they do is just return what another function returns" -- I don't think that's awkward.

I would add _splGetSslSrv/_splGetManuSrv, and then add the getters as I described that just forward to the _splGet*Srvs.

Pinging @fincs / @yellows8 about this though.

Reply:
^sure

Reply:
Hopefully I did it correctly with the latest commit